drivers/gpio/gpio_ite_it8xxx2_v2.c

```diff
@@ -380,6 +382,8 @@ static void gpio_ite_isr(const void *arg)
 	uint8_t num_pins = gpio_config->num_pins;
 	uint8_t pin;
 
+	__ASSERT(num_pins <= ITE_GPIO_MAX_PINS, "The maximum number of pins per port is 8.");
```

Comment:
I'd suggest converting this to a `BUILD_ASSERT()` where this is retrieved from the devicetree (`.num_pins = DT_INST_PROP(inst, ngpios),` further down this file).

Reply:
Done.